openslides_backend/migrations/migrations/0062_unset_presence_of_removed_users.py

```diff
@@ -0,0 +1,63 @@
+from collections import defaultdict
+
+from datastore.migrations import BaseModelMigration
+from datastore.writer.core import BaseRequestEvent, RequestUpdateEvent
+
+from openslides_backend.shared.patterns import fqid_from_collection_and_id
+
+
+class Migration(BaseModelMigration):
+    """
+    This migration adds the user_id "-1" to all existing action_workers.
+    This is the number usually used for calls using the internal route.
+    """
+
+    target_migration_index = 63
+
+    def migrate_models(self) -> list[BaseRequestEvent] | None:
+        present_users_per_meeting: dict[int, list[int]] = defaultdict(list)
+        meetings_per_present_user: dict[int, list[int]] = defaultdict(list)
+        meetings = self.reader.get_all(
+            "meeting", ["group_ids", "present_user_ids", "meeting_user_ids"]
+        )
+        users = self.reader.get_all(
+            "user", ["is_present_in_meeting_ids", "meeting_user_ids"]
+        )
```

Comment:
Ui, that is a large operation, getting 1000+ meetings and 10000+ users in worse case. Use get_all with cautious
and use filter, etc instead.

Reply:
I thought about this but I couldn't think of a solution, because I have to compare in both directions if the presence relation between the meeting and user is corrupted. If only one way was necessary I could at least filter for all the users that are present in meetings.

Reply:
Due to refactoring this does not exist anymore. Now I'm using one `filter` for the meeting users and `get_many` for meeting and users.